src/util.jl

```diff
@@ -42,10 +42,12 @@ end
 
 """returns the DCT filters"""
 function dct(nfilters::Int, ninput::Int)
-    basis = Array(Float32, nfilters, ninput)
+    basis = Array{Float32}(undef, nfilters, ninput)
     samples = (1f0:2f0:2ninput) * π / 2ninput
     for i = 1:nfilters
-        basis[i, :] = cos(i * samples)
+        for j in 1:length(samples)
+            basis[i, j] = cos(i * samples[j])
+        end
```

Comment:
basis[i, :] = cos.(i .* samples)

Reply:
it is the preferred way to do it, as it allows loop fusion (here it doesn't apply, but better to follow the best style)

Reply:
Thanks for explanation!

Reply:
nevertheless, the for loop version is guaranteed to be faster.